public/js/actions/breakpoints.js

```diff
@@ -22,15 +23,51 @@ function enableBreakpoint(location: Location) {
   return addBreakpoint(location);
 }
 
+function isOriginalLocation(state, location: Location) {
+  const source = getSource(state, location.sourceId);
+  return isOriginal(source.toJS());
+}
+
+function getGeneratedLocation(state, location: Location) {
+  const source = getSource(state, location.sourceId);
+  if (isOriginal(source.toJS())) {
+    return getGeneratedSourceLocation(source.toJS(), location);
+  }
+
+  return location;
+}
+
 function _breakpointExists(state, location: Location) {
+  location = getGeneratedLocation(state, location);
   const currentBp = getBreakpoint(state, location);
   return currentBp && !currentBp.get("disabled");
 }
 
 function _getOrCreateBreakpoint(state, location, condition) {
+  location = getGeneratedLocation(state, location);
   return getBreakpoint(state, location) || fromJS({ location, condition });
 }
 
+function getBreakpointText({
+  breakpoint, getState, getTextForLine, actualLocation, location }) {
```

Comment:
I'm a little confused between the difference of `actualLocation` and `location` here. It looks like you pass in the before/after result of the `setBreakpoint` call, and `actualLocation` is only different if it slid the breakpoint. It doesn't have to do with sourcemapping.